pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -684,27 +687,85 @@ func newOauthClient(tokenSource oauth2.TokenSource) (*http.Client, error) {
 func (manager *GCEServiceManager) CreateDisk(
 	project string,
 	zone string,
-	disk *compute.Disk) (*compute.Operation, error) {
+	disk gceObject) (gceObject, error) {
+
+	switch v := disk.(type) {
+	case *computealpha.Disk:
+		return manager.gce.serviceAlpha.Disks.Insert(project, zone, disk.(*computealpha.Disk)).Do()
+	case *computebeta.Disk:
+		return manager.gce.serviceBeta.Disks.Insert(project, zone, disk.(*computebeta.Disk)).Do()
+	case *compute.Disk:
+		return manager.gce.service.Disks.Insert(project, zone, disk.(*compute.Disk)).Do()
+	default:
+		return nil, fmt.Errorf("unexpected type: %T", v)
+	}
+}
+
+func (manager *GCEServiceManager) AttachDisk(
+	project string,
+	zone string,
+	instanceName string,
+	attachedDisk gceObject) (gceObject, error) {
 
-	return manager.gce.service.Disks.Insert(project, zone, disk).Do()
+	switch v := attachedDisk.(type) {
```

Comment:
This function uses switch statements, but the others use if/else.  Can it be consistent?